proxy/src/control/destination/mod.rs

```diff
@@ -88,11 +88,25 @@ pub struct Resolution<B> {
 
 /// Metadata describing an endpoint.
 #[derive(Clone, Debug, Hash, Eq, PartialEq)]
-struct Metadata {
+pub struct Metadata {
     /// A set of Prometheus metric labels describing the destination.
-    metric_labels: Option<DstLabels>,
+    dst_labels: Option<DstLabels>,
+
+    /// How to verify TLS for the endpoint.
+    tls_identity: Option<TlsIdentity>,
+}
+
+/// How to verify TLS for an endpoint.
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
```

Comment:
I understand that we have to be `Hash`, `PartialEq`, and `Eq` because that's what `Metadata` is. But IMO it would be better to avoid adding these if we can, so that we don't have to think about what equality means for domain names, etc. (considering it's case-insensitive).

Reply:
I agree, but I thought this was better than having `Metadata` ignore whether the TLS identity is the same when checking for equality.